### DISCUSSION TITLE
wire;model for file input type not updating temporaryUrl when setting file with $wire.set from alpine js

### DISCUSSION THREAD

**Discussion by Shashank-Bhatt1** [livewire/livewire#7998] [7. Bugs (archived)] [answered]
### Livewire version

v2.11

### Laravel version

v10.45.0

### Which PHP version are you using?

PHP 8.2

### Repository or Wirebox URL

https://wirebox.app/b/4z198

### Additional information

this is blade file component side where trying to set uploadedImage from $wire.set of alpine had no effect on temporaryUrl being updated. 

```
<div x-data @image-processed.window="($event) => {
                        $wire.set('uploadedImage', $event.detail.bundleFile);
}">
@if ($uploadedImage)
       <img class="product-view-details img-fluid" src="{{ $uploadedImage->temporaryUrl() }}"
              alt="temporary image">
@endif

<div style="margin-top:15px; opacity:0;">
       <input wire:model="uploadedImage" style="width: 90%" id="product_image" type="file"
              class="@error('uploadedImage') is-invalid @enderror">
</div>
</div>
```

In Above wirebox url, temporaryUrl could not work as i am not familiar muh about php/laravel so could not make it work.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@Shashank-Bhatt1 you need to use `$wire.upload()`. See here for details https://livewire.laravel.com/docs/uploads#javascript-upload-api

**Reply by Shashank-Bhatt1** [reply]
Hi @joshhanley Thanks for the prompt response but i have question whether this method directly saves photo or let us preview same as `temporaryUrl()` does?

**Reply by joshhanley** [reply, marked as answer]
@Shashank-Bhatt1 no worries! `$wire.upload()` uses the same API as if you put `wire:model` on your file input, so it will create a `TemporaryUploadedFile`, which has the `temporaryUrl()` method.

**Reply by Shashank-Bhatt1** [reply]
@joshhanley Yes it worked perfectly as needed. Thanks. :)

**Reply by joshhanley** [reply]
@Shashank-Bhatt1 sweet! No worries! 🙂
